src/battle_transition.c

```diff
@@ -0,0 +1,2513 @@
+{
+    .tileTag = 0xFFFF,
+    .paletteTag = 4106,
+    .oam = &gOamData_83FD9A4,
+    .anims = sSpriteAnimTable_83FD9C4,
+    .images = sSpriteImageTable_83FD9B4,
+    .affineAnims = gDummySpriteAffineAnimTable,
+    .callback = sub_811C90C
+};
+
+static const u16 gFieldEffectObjectPalette10[] = INCBIN_U16("graphics/field_effect_objects/palettes/10.gbapal");
+
+const struct SpritePalette gFieldEffectObjectPaletteInfo10 =
+{
+    gFieldEffectObjectPalette10, 0x1009
+};
+
+static const u16 sMugshotPal_Sydney[] = INCBIN_U16("graphics/battle_transitions/sidney_bg.gbapal");
+static const u16 sMugshotPal_Phoebe[] = INCBIN_U16("graphics/battle_transitions/phoebe_bg.gbapal");
+static const u16 sMugshotPal_Glacia[] = INCBIN_U16("graphics/battle_transitions/glacia_bg.gbapal");
+static const u16 sMugshotPal_Drake[] = INCBIN_U16("graphics/battle_transitions/drake_bg.gbapal");
+static const u16 sMugshotPal_Steven[] = INCBIN_U16("graphics/battle_transitions/steven_bg.gbapal");
+static const u16 sMugshotPal_Brendan[] = INCBIN_U16("graphics/battle_transitions/brendan_bg.gbapal");
+static const u16 sMugshotPal_May[] = INCBIN_U16("graphics/battle_transitions/may_bg.gbapal");
+
+static const u16 * const sOpponentMugshotsPals[MUGSHOTS_NO] =
+{
+    sMugshotPal_Sydney,
+	sMugshotPal_Phoebe,
+	sMugshotPal_Glacia,
+	sMugshotPal_Drake,
+	sMugshotPal_Steven
+};
+
+static const u16 * const sPlayerMugshotsPals[2] =
+{
+    sMugshotPal_Brendan,
+    sMugshotPal_May
+};
+
+static const u16 sUnusedTrainerPalette[] = INCBIN_U16("graphics/battle_transitions/unused_trainer.gbapal");
+static const struct SpritePalette sSpritePalette_UnusedTrainer =
+{
+    sUnusedTrainerPalette, 0x100A
+};
+
+static const u16 sBigPokeball_Tilemap[] = INCBIN_U16("graphics/battle_transitions/big_pokeball_map.bin");
+static const u16 sMugshotsTilemap[] = INCBIN_U16("graphics/battle_transitions/elite_four_bg_map.bin");
+
+// actual code starts here
+
+void sub_811AABC(u8 transitionID)
+{
+    gMain.callback2 = sub_8054398;
+    LaunchBattleTransitionTask(transitionID);
+}
+
+void sub_811AAD8(u8 transitionID)
+{
+    LaunchBattleTransitionTask(transitionID);
+}
+
+#define tState          data[0]
+#define tTransitionID   data[1]
+#define tTransitionDone data[15]
+
+bool8 IsBattleTransitionDone(void)
+{
+    u8 taskID = FindTaskIdByFunc(Task_BattleTransitionMain);
+    if (gTasks[taskID].tTransitionDone)
+    {
+        DestroyTask(taskID);
+        return TRUE;
+    }
+    else
+        return FALSE;
+}
+
+static void LaunchBattleTransitionTask(u8 transitionID)
+{
+    u8 taskID = CreateTask(Task_BattleTransitionMain, 2);
+    gTasks[taskID].tTransitionID = transitionID;
+}
+
+static void Task_BattleTransitionMain(u8 taskID)
+{
+    while (sMainTransitionPhases[gTasks[taskID].tState](&gTasks[taskID]));
+}
+
+static bool8 Transition_Phase1(struct Task* task)
+{
+    sub_807DE10();
+    CpuSet(gPlttBufferFaded, gPlttBufferUnfaded, 0x4000100);
```

Comment:
`CpuCopy32(gPlttBufferFaded, gPlttBufferUnfaded, 0x400);` Please don't use `CpuSet` directly since we have macros that call it.